ynput/ayon-core
# Publisher: Use hierarchy model

## Changelog Description
Publisher is using `HierarchyModel` to get folders and tasks.

## Additional info
First modification of publisher tool before using AYON entities. Biggest change is that `HierarchyModel` is used to query folders. There were some necessary changes in UI to create wrapper controllers for context selection in create page and asset dialog. Use folder naming in publisher tool. Fixed few found bugs in Hierarchy model and added more public methods to `FoldersWidget` and `TasksWidget`.

## Testing notes:
1. Launch host of your choice. Ideally not traypublisher...
2. Open publisher tool.
3. Page on Create tab should show context and is responsible to folder and task changes.
4. Existing instances should not have any issues with context.
5. Newly created instances should have correct context selected in UI.
6. It is possible to change context of instance in Publish tab.
7. Instance with changed context can be published.
